src/jenesis/keyring/__init__.py

```diff
@@ -15,23 +16,35 @@
     pass
 
 
-def query_keychain_item(name: str) -> Union[LocalInfo, OfflineInfo]:
-    system = platform.system()
-    if system == "Darwin":
-        return macos.query_keychain_item(name)
+def query_keychain_item(name: str, backend: str = "os") -> Union[LocalInfo, OfflineInfo]:
+    if backend == "os":
+        system = platform.system()
+        if system == "Darwin":
+            return macos.query_keychain_item(name)
 
-    if system == "Linux":
-        return linux.query_keychain_item(name)
+        if system == "Linux":
+            return linux.query_keychain_item(name)
 
-    raise RuntimeError(f"Platform {system} not currently supported")
+        raise RuntimeError(f"Platform {system} not currently supported")
 
+    if backend == "test":
+        return test.query_keychain_item(name)
 
-def query_keychain_items() -> List[str]:
-    system = platform.system()
-    if system == "Darwin":
-        return macos.query_keychain_items()
+    raise RuntimeError(f"Keyring backend '{backend}' not currently supported")
 
-    if system == "Linux":
-        return linux.query_keychain_items()
 
-    raise RuntimeError(f"Platform {system} not currently supported")
+def query_keychain_items(backend: str = "os") -> List[str]:
+    if backend == "os":
```

Comment:
Good point. There aren't any plans to support other keyring backends at the moment, but if we expand it would definitely be better to add a backend class.